redis/src/test/scala/zio/redis/KeysSpec.scala

```diff
@@ -5,16 +5,38 @@ import zio.duration._
 import zio.test.Assertion._
 import zio.test.TestAspect._
 import zio.test._
+//import zio.{Chunk, ZIO}
+import java.time.{Duration => JavaDuration}
+// import SecondRedisExecutorLayer._
+// import zio.redis.Output.OptionalOutput
+// import zio.redis.Output.MultiStringOutput
+// import zio.redis.RedisError.ProtocolError
+//import zio.Has
 import zio.test.environment.{ TestClock, TestConsole, TestRandom, TestSystem }
 import zio.{ Chunk, Has, ZIO }
+import zio.ZLayer
+import zio.logging.Logging
+// import zio.redis.ApiSpec.RedisExecutorWrapper
+// import zio.redis.Output.OptionalOutput
+// import zio.redis.Output.MultiStringOutput
+// import zio.redis.ApiSpec.RedisExecutorWrapper
+// import zio.redis.Output.OptionalOutput
+// import zio.redis.Output.MultiStringOutput
 
 trait KeysSpec extends BaseSpec {
 
   val keysSuite: Spec[Has[Clock.Service] with Has[RedisExecutor.Service] with Has[TestClock.Service] with Has[
     TestConsole.Service
   ] with Has[TestRandom.Service] with Has[TestSystem.Service] with Has[RedisExecutor.Service] with Has[
     Annotations.Service
-  ], TestFailure[RedisError], TestSuccess] =
+  ], TestFailure[RedisError], TestSuccess] = {
+    final case class RedisExecutorWrapper(re: RedisExecutor.Service)
+
+    val secondConfigLayer = ZLayer.succeed(RedisConfig("localhost", 6380))
+    val secondRedisService = (Logging.ignore ++ secondConfigLayer >>> RedisExecutor.live).fresh.project(service => RedisExecutorWrapper(service))
```

Comment:
You just need this:
```suggestion
    val secondRedisService = (Logging.ignore ++ secondConfigLayer) >>> RedisExecutor.live
```
Then, instead of `provide` use `provideLayer` method in your test and give it `secondRedisService`.

Reply:
Ah nice, thanks, still haven't figured out the differences between all the different provides :)